lib/ssl/test/openssl_stapling_SUITE.erl

```diff
@@ -170,24 +213,44 @@ stapling_helper(Config, Opts) ->
     Data = ssl_test_lib:check_active_receive(Client, Data),
     ssl_test_lib:close(Server),
     ssl_test_lib:close(Client).
+
 %%--------------------------------------------------------------------
-stapling_revoked() ->
+staple_not_designated() ->
+    [{doc,"Verify OCSP stapling works without nonce."
+      "Response signed with certificate issued directly by issuer of server "
+      "certificate but not  designated for OCSP signing (extKeyUsage missing "
+      "OCSP signing)."}].
+staple_not_designated(Config)
+  when is_list(Config) ->
+    stapling_negative_helper(Config, "a.server/cacerts.pem",
+                                  openssl_ocsp, bad_certificate).
+
+staple_wrong_issuer() ->
+    [{doc,"Verify OCSP stapling works without nonce."
+      "Response signed with certificate not related to issuer of server "
+      "certificate."}].
+staple_wrong_issuer(Config)
+  when is_list(Config) ->
+    stapling_negative_helper(Config, "a.server/cacerts.pem",
+                                  openssl_ocsp, bad_certificate).
+
+cert_status_revoked() ->
     [{doc, "Verify OCSP stapling works with revoked certificate."}].
-stapling_revoked(Config)
+cert_status_revoked(Config)
   when is_list(Config) ->
     stapling_negative_helper(Config, "revoked/cacerts.pem",
                                   openssl_ocsp_revoked, certificate_revoked).
 
-stapling_undetermined() ->
+cert_status_undetermined() ->
     [{doc, "Verify OCSP stapling works with certificate with undetermined status."}].
-stapling_undetermined(Config)
+cert_status_undetermined(Config)
   when is_list(Config) ->
     stapling_negative_helper(Config, "undetermined/cacerts.pem",
                                   openssl_ocsp_undetermined, bad_certificate).
 
-stapling_no_staple() ->
+staple_missing() ->
```

Comment:
@u3s - check (at least by code reading) how ssl client behaves if it receives OCSP response which wasn't requested.